### DISCUSSION TITLE
How to stop changing NvimTree root?

### DISCUSSION THREAD

**Discussion by lai32290** [nvim-tree/nvim-tree.lua#1943] [Q&A]
When I open my nvim in folder `MyProject` and open the NvimTree (`:NvimTreeToggle`), it uses `MyProject` as root.

But when I `gf` to some import folder, like:

```
import { MyComponente } from './FolderWith2Files'
```

It shows the `netrw` and let me select the file that I wanna open, since that, when I open NvimTree again, the root become the `FolderWith2Files` instead of `MyProject`, but `:pwd` is still in `MyProject` and `Telescope` is also still looking to `MyProject`.

How can I make it stop changing root for NvimTree when I `gf` to a folder?

**Comment by alex-courtis** [comment]
> How can I make it stop changing root for NvimTree when I `gf` to a folder?

Is `gf` a netrw command? Does it change vim's working directory?

Do you have `nvim-tree.sync_root_with_cwd` set? That will cause nvim-tree to change it's root when vim's cwd is changed.

Naive suggestion: use nvim-tree instead of netrw to select files etc.

**Reply by lai32290** [reply]
Is gf a netrw command? Does it change vim's working directory?
A: oh, sorry, actually it shows nvim-tree, I'm not using `netrw` as I know.

Do you have nvim-tree.sync_root_with_cwd set? That will cause nvim-tree to change it's root when vim's cwd is changed.
A: I don't think that I have this config set, at least it's not in my `.setup({`

I recorded a video here to demonstrate what is happening here, hope it can be more clear than my explanation 😅


https://user-images.githubusercontent.com/903193/215797298-2fc5f245-43c0-44d3-b32a-3b38cafafed8.mov

**Reply by alex-courtis** [reply]
I think it is netrw: `:help netrw-gf`

That will open the directory in netrw, which is hijacked by nvim-tree. Set `hijack_netrw = false` to demonstrate this.

This is working as intended: you are effectively opening a directory in nvim-tree.

You could `gf` to the file itself, rather than the containing directory, which will open a new file buffer as you might expect.
